src/frontend/styles/_typography.scss

```diff
@@ -0,0 +1,140 @@
+/*
+with 1.2 scale ratio and 16px root font size
+name px rem
+xs	11.11	0.69
+sm	13.33	0.83
+md	16.00	1.00
+lg	19.20	1.20
+xl	23.04	1.44
+2xl	27.65	1.73
+3xl	33.18	2.07
+4xl	39.81	2.49
+5xl	47.78	2.99
+6xl	57.33	3.58
+*/
+
+@mixin typeSystem {
+  --default-primary-font-family: 'Rubik', sans-serif;
+  --default-secondary-font-family: 'Cabin', sans-serif;
+  --primary-font-family: var(--default-primary-font-family);
+  --secondary-font-family: var(--default-secondary-font-family);
+  --light: 300;
+  --regular: 400;
+  --medium: 500;
+  --semibold: 600;
+  --bold: 700;
+  --extrabold: 900;
+
+  /* set base values */
+  --text-scale-ratio: 1.2;
+  font-size: 16px;
+
+  /* type scale */
+  --text-xs: calc(1rem / (var(--text-scale-ratio) * var(--text-scale-ratio)));
+  --text-sm: calc(1rem / var(--text-scale-ratio));
+  --text-md: calc(1rem);
+  --text-lg: calc(1rem * var(--text-scale-ratio));
+  --text-xl: calc(1rem * var(--text-scale-ratio) * var(--text-scale-ratio));
+  --text-2xl: calc(
+    1rem * var(--text-scale-ratio) * var(--text-scale-ratio) *
+      var(--text-scale-ratio)
+  );
+  --text-3xl: calc(
+    1rem * var(--text-scale-ratio) * var(--text-scale-ratio) *
+      var(--text-scale-ratio) * var(--text-scale-ratio)
+  );
+  --text-4xl: calc(
+    1rem * var(--text-scale-ratio) * var(--text-scale-ratio) *
+      var(--text-scale-ratio) * var(--text-scale-ratio) *
+      var(--text-scale-ratio)
+  );
+  --text-5xl: calc(
+    1rem * var(--text-scale-ratio) * var(--text-scale-ratio) *
+      var(--text-scale-ratio) * var(--text-scale-ratio) *
+      var(--text-scale-ratio) * var(--text-scale-ratio)
+  );
+  --text-6xl: calc(
+    1rem * var(--text-scale-ratio) * var(--text-scale-ratio) *
+      var(--text-scale-ratio) * var(--text-scale-ratio) *
+      var(--text-scale-ratio) * var(--text-scale-ratio) *
+      var(--text-scale-ratio)
+  );
```

Comment:
is there a reason for the 3 different styles of defining these variables?

there's the ones above this comment with a fixed `1rem` multiplied multiple times by (in this case) 1.2.

There's this with just fixed values (like the multiplication was already done and added here)

```
  /* spacing values */
  --space-3xs: 0.25em;
  --space-2xs: 0.375em;
  --space-xs: 0.5em;
  --space-sm: 0.75em;
  --space-md: 1em;
  --space-lg: 1.5em;
  --space-xl: 3em;
  --space-2xl: 4.5em;
  --space-3xl: 8.5em;
```

And there's this, where the multiplication is done with a fixed multiplied and the now the variable part is the `1rem`.

```
  --space-unit-fixed: 1rem;
  --space-2xs-fixed: calc(0.25 * var(--space-unit-fixed));
  --space-xs-fixed: calc(0.5 * var(--space-unit-fixed));
  --space-sm-fixed: calc(0.75 * var(--space-unit-fixed));
  --space-md-fixed: calc(1 * var(--space-unit-fixed));
  --space-lg-fixed: calc(1.2 * var(--space-unit-fixed));
  --space-xl-fixed: calc(2.5 * var(--space-unit-fixed));
  --space-2xl-fixed: calc(4.5 * var(--space-unit-fixed));
```

I wonder if we really need this flexibility and complexity or if we can just hardcode some values that work for what we need, I can't imagine this changing that often but maybe I'm missing the reason for these differences.

Reply:
I responded in detail below. To reiterate my overall point of view, I think this reduces complexity dramatically (defined in one top level styles folder vs hard coded in each component), allows the app's frontend to be maintained and iterated upon easily, and helps encourage frontend components to use standardized values so that we can have a visually consistent UX while meeting the needs of each component's design criteria.